### PR TITLE
Updated Azure discovery docs after move to seperate package

### PR DESCRIPTION
**Pull request by afscrome**
Signed-off-by: Alex Crome <afscrome@users.noreply.github.com>

## Hey, I just made a Pull Request!

Updated the Azure Discovery docs to work with recent package changes to extract `plugin-catalog-backend-module-azure` to it's own package.

This change mostly replicates the change from the github discovery doc, with appropriate changes for Azure.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/integrations/azure/discovery.md [hunk 1]
```diff
@@ -12,6 +12,34 @@ DevOps organization and register entities matching the configured path. This can
...
+
+```diff
+// In packages/backend/src/plugins/catalog.ts
++import { AzureDevOpsDiscoveryProcessor } from '@backstage/plugin-catalog-backend-module-azure';
+
+ export default async function createPlugin(
+   env: PluginEnvironment,
+ ): Promise<Router> {
+   const builder = await CatalogBuilder.create(env);
++  builder.addProcessor(AzureDevOpsDiscoveryProcessor.fromConfig(env.config, { logger: env.logger }));
```
Comment (afscrome) [inline docs/integrations/azure/discovery.md:38]: should this use `config` and `logger` as opposed to `env.config` and `env.logger` like the other docs?  Without the `env`, I couldn't get them working in my instance (created with `npx @backstage/create-app`)

* https://backstage.io/docs/integrations/github/discovery
*  https://github.com/backstage/backstage/releases/tag/v0.71.0

File: docs/integrations/azure/discovery.md [hunk 2]
```diff
@@ -12,6 +12,34 @@ DevOps organization and register entities matching the configured path. This can
 be useful as an alternative to static locations or manually adding things to the
 catalog.
 
+## Installation
+
+You will have to add the processors in the catalog initialization code of your
+backend. They are not installed by default, therefore you have to add a
+dependency to `@backstage/plugin-catalog-backend-module-github` to your backend
```
Comment (freben) [inline docs/integrations/azure/discovery.md:19]: typo: github -> azure

File: docs/integrations/azure/discovery.md (hunk 1 above)
Comment (freben) [reply docs/integrations/azure/discovery.md:38]: No keep it like this, I'm overhauling all of these plugins to have the exact same (explicit but easier to understand and copy-paste) structure

https://github.com/backstage/backstage/pull/10150

**Review by freben — COMMENTED** [review]
Thank you! One minor typo to fix

File: docs/integrations/azure/discovery.md (hunk 2 above)
Comment (afscrome) [reply docs/integrations/azure/discovery.md]: whoops, my turn for a copy & paste error. Thanks for catching.

**Review by freben — APPROVED** [review]

**Pull request merged**